# origin: bk
# cset: 1.1493.3.10
# inclusion: n/a (projected: 2.4.28-pre3)
# description: fix hw autoneg problems
# revision date: 2004-09-03

# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2004/08/30 15:12:10-07:00 mchan@broadcom.com 
#   [TG3]: Check MAC_STATUS_SIGNAL_DET in serdes polling.
#   
#   Otherwise we do not handle properly the case
#   where the switch/hub does not support auto-
#   negotiation.  This is what was breaking 5704
#   hw fiber autoneg.
#   
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
# drivers/net/tg3.c
#   2004/08/30 15:11:48-07:00 mchan@broadcom.com +2 -1
#   [TG3]: Check MAC_STATUS_SIGNAL_DET in serdes polling.
#   
#   Otherwise we do not handle properly the case
#   where the switch/hub does not support auto-
#   negotiation.  This is what was breaking 5704
#   hw fiber autoneg.
#   
#   Signed-off-by: David S. Miller <davem@davemloft.net>
# 
diff -Nru a/drivers/net/tg3.c b/drivers/net/tg3.c
--- a/drivers/net/tg3.c	2004-09-03 18:33:38 -07:00
+++ b/drivers/net/tg3.c	2004-09-03 18:33:38 -07:00
@@ -5602,7 +5602,8 @@
 				need_setup = 1;
 			}
 			if (! netif_carrier_ok(tp->dev) &&
-			    (mac_stat & MAC_STATUS_PCS_SYNCED)) {
+			    (mac_stat & (MAC_STATUS_PCS_SYNCED |
+					 MAC_STATUS_SIGNAL_DET))) {
 				need_setup = 1;
 			}
 			if (need_setup) {
